File: C++/MiniMaxSum.cpp
#include <iostream>
#include <vector>

void miniMaxSum(std::vector<int> arr) {
	long total = 0, min, max=arr[0];
	for(int i = 0; i < arr.size(); i++) {
		std::cout << "total:" <<total << " + arr[" << i << "]: " <<arr[i] << "="  << total+arr[i] << std::endl;
		total = total +  arr[i];
	}
	std::cout << "TOTAL: " <<total<<std::endl; 
	for(int i=0; i < arr.size(); i++) {
		if(min > total - arr[i]) {
			min = total - arr[i];
		}
		if(max < total - arr[i]) {
			max = total - arr[i];
		}
	}
	std::cout << min << " " << max;
}

int main() {
	const int MAX_NUM = 5;
	int temp;
	std::vector<int> arr;
	for(int i=0;i<MAX_NUM;i++) {
		std::cin >> temp;
		arr.push_back(temp);
	}

	miniMaxSum(arr);
	return 0;
}

File: java/MiniMaxSum.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class MiniMaxSum {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    List<Integer> arr =  new ArrayList<>();
    for(int i = 0; i < 5 ; i ++) {
      arr.add(in.nextInt());
    }
    miniMaxSum(arr);
    in.close();
  }

  public static void miniMaxSum(List<Integer> arr) {
    // Write your code here
        long min = Long.MAX_VALUE, max = Long.MIN_VALUE, total = 0;
        
        for(Integer num : arr) {
            total += num;
        }
        for(Integer num : arr ) {
          System.out.println("TEST:" + (total-num));
          System.out.println("MIN:" + min);
          
            if(max < total-num) {
                max = total-num;
            }if(min > total-num) {
                min = total-num;
            }
        }
        System.out.print(min+" ");
        System.out.print(max);
    }
}